dynamax/hidden_markov_model/inference_test.py

```diff
@@ -285,3 +286,32 @@ def test_parallel_smoother(key=0, num_timesteps=100, num_states=3):
     posterior = core.hmm_smoother(initial_probs, transition_matrix, log_likelihoods)
     posterior2 = parallel.hmm_smoother(initial_probs, transition_matrix, log_likelihoods)
     assert jnp.allclose(posterior.smoothed_probs, posterior2.smoothed_probs, atol=1e-1)
+
+
+def test_parallel_posterior_sample(
```

Comment:
I believe we commented this test out for a few reasons:
1. it's expensive to draw so many samples
2. it's a randomized test, so it can fail with some probability
3. we could do a little math to figure out the failure probability as a function of `eps` and `num_samples`, but we never took the time to do so.
I think this is exactly the right test to run to check for correctness though, and I'd love to have it in our rotation as long as it reliably passes!

Reply:
On a practical level, I assume it would be reliable because the seed is fixed?